### PR TITLE
the getchfactor

### PR DESCRIPTION
**Pull request by getchoo**

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -91,3 +118,14 @@ async fn setup(
...
+
+async fn handle_shutdown(shard_manager: Arc<serenity::ShardManager>, reason: ShutdownReason) {
+	let reason = match reason {
+		ShutdownReason::CtrlC => "Interrupted",
+		ShutdownReason::Sigterm => "Received SIGTERM",
+	};
+
+	warn!("{reason}! Shutting down bot...");
+	shard_manager.shutdown_all().await;
+	println!("Everything is shutdown. Goodbye!");
```
Comment (Scrumplex) [inline src/main.rs:130]: why not `info!`?

**Review by Scrumplex — APPROVED** [review]
Changes LGTM

File: src/main.rs (hunk 1 above)
Comment (getchoo) [reply src/main.rs:130]: It's just to bring attention to it

**Review by TheKodeToad — APPROVED** [review]
I did deliberately do context for legibility but ig ctx is well known enough and means ~~no need for ctx: context~~ (nvm I was thinking of something else?)

File: src/main.rs (hunk 1 above)
Comment (TheKodeToad) [reply src/main.rs:130]: Best show this even if info messages are hidden

**Comment by TheKodeToad** [comment]
~~Probably should have tested these changes before merging :trollface: :trollface:~~

**Comment by TheKodeToad** [comment]
It's still working! :tada: Thanks

**Pull request merged**